packages/flutter/lib/src/material/paginated_data_table.dart

```diff
@@ -84,8 +84,11 @@ class PaginatedDataTable extends StatefulWidget {
     assert(sortAscending != null);
     assert(rowsPerPage != null);
     assert(rowsPerPage > 0);
-    assert(availableRowsPerPage != null);
-    assert(availableRowsPerPage.contains(rowsPerPage));
+    assert(() {
+      if (onRowsPerPageChanged == null)
+        return true;
+      return availableRowsPerPage != null && availableRowsPerPage.contains(rowsPerPage);
```

Comment:
flip this around, like:
```dart
assert(() {
  if (onRowsPerPageChanged != null)
    assert(availableRowsPerPage != null && availableRowsPerPage.contains(rowsPerPage));
  return true;
});
```

...otherwise when the assert throws, it'll include all three lines of code.